core/io/file_access_compressed.cpp

```diff
@@ -37,9 +37,7 @@ void FileAccessCompressed::configure(const String &p_magic, Compression::Mode p_
 	if (magic.length() > 4) {
 		magic = magic.substr(0, 4);
 	} else {
-		while (magic.length() < 4) {
-			magic += " ";
-		}
+		magic += String(" ").repeat(4 - magic.length());
```

Comment:
Shouldn't this use `rpad()`?

Reply:
I changed the whole thing to `magic = (magic + "    ").substr(0, 4)`